src/Spectre.libClustering/Partition.cpp

```diff
@@ -0,0 +1,77 @@
+/*
+* Partition.h
+* Provides class implementation that simplifies partition vectors.
+*
+Copyright 2017 Dariusz Kuchta
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "Partition.h"
+#include <exception>
+
+namespace Spectre::libClustering
+{
+	Partition::Partition(gsl::span<unsigned int> partition) : m_Partition(partition.size())
```

Comment:
From @gmrukwa : please add "explicit" keyword to the constructor to avoid implicit conversions.